server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java

```diff
@@ -260,13 +271,14 @@ private void createTemplateWithinZone(Long zId, TemplateProfile profile, VMTempl
                 }
 
                 // Check if zone is disabled
-                if (Grouping.AllocationState.Disabled == zone.getAllocationState()) {
+                if (Grouping.AllocationState.Disabled == zone.getAllocationState() && !template.getTemplateType().equals(Storage.TemplateType.SYSTEM)) {
                     s_logger.info("Zone " + zoneId + " is disabled. Skip downloading template to its image store " + imageStore.getId());
                     continue;
                 }
```

Comment:
So we should download systemVM templates to a disabled zone? I don't see the exception. We can download it before disabling or after enabling. Why does the reason for not allowing it for normal templates not apply to systemVM templates?